### PR TITLE
Add signedness semantics for integer types

### PR DESCRIPTION
**Pull request by math-fehr**
This PR adds signedness semantics for integer types, as MLIR does.
This PR also fix one test that was failing, but not running on the CI since the MLIR tests are not ran on the CI.
(Note, I should fix in a following PR this, since most of the MLIR files do not need MLIR anymore).

### CODE REVIEW

File: xdsl/parser.py [hunk 1]
```diff
@@ -123,6 +123,10 @@ def __post_init__(self):
         else:
             self._pos = Position(self.str)
 
+    def get_pos(self) -> Position | None:
```
Comment (webmiche) [inline xdsl/parser.py:126]: How is this change related to this PR? It does not seem to get used, in contrast, it is not used where it could (Line 869).

File: tests/filecheck/mlir-conversion/builtin_attrs.mlir [hunk 2]
```diff
@@ -31,19 +44,19 @@
 
   "func.func"() ({}) {function_type = () -> (), value = true, sym_name = "true_attr"} : () -> ()
 
-  // CHECK: value = true
+  // CHECK: "value" = true
```
Comment (webmiche) [inline tests/filecheck/mlir-conversion/builtin_attrs.mlir:47]: I am confused about this change. Did this work before? Why should this be changed in this PR?

It feels like printing as xDSL should have this result, whereas printing with MLIR should be the old one :sweat_smile:

File: tests/filecheck/mlir-conversion/builtin_attrs.mlir [hunk 3]
```diff
@@ -80,19 +93,17 @@
   // CHECK: (tensor<4xf32>, tensor<f32>, tensor<1x12xi32>, tensor<*xf64>)
 
   "func.func"() ({}) {function_type = () -> (),
-                      value1 = dense<0> : tensor<1xi32>,
+                      value1 = dense<[0]> : tensor<1xi32>,
```
Comment (webmiche) [inline tests/filecheck/mlir-conversion/builtin_attrs.mlir:96]: Unrelated?

File: tests/filecheck/parser-printer/builtin_types.xdsl [hunk 4]
```diff
@@ -2,14 +2,15 @@
 
 
 builtin.module() {
-  func.func() ["sym_name" = "tuple_to_tuple", "function_type" = !fun<[!tuple<[!i32]>], [!tuple<[!i32]>]>, "sym_visibility" = "private"] {
-    ^0(%0 : !tuple<[!i32]>):
-      func.return(%0 : !tuple<[!i32]>)
-  }
-}
+  func.func() ["sym_name" = "tuple_to_tuple", "value" = !fun<[!tuple<[!i32]>], [!tuple<[!i32]>]>, "function_type" = !fun<[], []>] {}
+  // CHECK: "value" = !fun<[!tuple<[!i32]>], [!tuple<[!i32]>]>
```
Comment (webmiche) [inline tests/filecheck/parser-printer/builtin_types.xdsl:6]: I don't quite get this change. AFAIU, you are using a function operation here to show that `!fun<[!tuple<[!i32]>], [!tuple<[!i32]>]>` can be parsed/printed. Why use a function operation? Aren't there easier operations to do that with? You can give `sym_name` and `value` to any operation to keep the name of the test around as well, right?

I feel like I am confused by these tests each time I see them. Would it make sense to have an empty `test.testing` operation just to make these tests easier to understand?

**Review by webmiche — APPROVED** [review]
LGTM, module some changes that feel unrelated to me. I generally feel these maybe should be their own PR, as it confused me a bit in this PR.

File: xdsl/parser.py (hunk 1 above)
Comment (math-fehr) [reply xdsl/parser.py:126]: I used it in the Signedness parser. Otherwise, I would get a warning since `_pos` is "private".

File: tests/filecheck/mlir-conversion/builtin_attrs.mlir (hunk 2 above)
Comment (math-fehr) [reply tests/filecheck/mlir-conversion/builtin_attrs.mlir:47]: I tried to described it in the PR message, but essentially this file was never running on the CI, and was actuallly not working.
So I fixed it to be able to test this PR. I'll fix the non-running issue on another PR though.
I put this change on another commit, so we won't have a single commit containing both changes (I'll not squash this time).

File: tests/filecheck/mlir-conversion/builtin_attrs.mlir (hunk 3 above)
Comment (math-fehr) [reply tests/filecheck/mlir-conversion/builtin_attrs.mlir:96]: See previous conversation

File: tests/filecheck/parser-printer/builtin_types.xdsl (hunk 4 above)
Comment (math-fehr) [reply tests/filecheck/parser-printer/builtin_types.xdsl:6]: Yeah, I'll add a `test.testing` in a following PR, that should make things better!

File: xdsl/parser.py (hunk 1 above)
Comment (webmiche) [reply xdsl/parser.py:126]: Ah yes, didn't see :+1:
